block-storage/src/main/scala/coop/rchain/blockstorage/util/io/package.scala

```diff
@@ -0,0 +1,76 @@
+package coop.rchain.blockstorage.util
+
+import java.io.IOException
+import java.nio.file._
+import java.util.stream.Collectors
+
+import cats.data.EitherT
+import cats.implicits._
+import cats.effect.Sync
+import coop.rchain.blockstorage.util.io.IOError.{IOErr, IOErrT}
+
+import scala.collection.JavaConverters._
+
+package object io {
+  private[io] def handleIo[F[_]: Sync, A](
+      io: => A,
+      handleIoException: IOException => IOError
+  ): F[IOErr[A]] =
+    Sync[F].delay {
```

Comment:
this change was from `Sync.delay.attempt` to `Sync.delay(try-catch)` - why is that?

Reply:
Because of this comment: https://github.com/rchain/rchain/pull/2160#discussion_r254684151

I hope you'll agree that it's still pure externally?